samza-core/src/main/java/org/apache/samza/storage/blobstore/util/BlobStoreUtil.java

```diff
@@ -90,6 +92,9 @@ public class BlobStoreUtil {
   private final SnapshotIndexSerde snapshotIndexSerde;
   private final RetryPolicyConfig retryPolicyConfig;
 
+  // LRU cache for blob ids that were TTL removed
+  private final Cache<String, Object> ttlRemovedBlobIdsCache = CacheBuilder.newBuilder().maximumSize(1000).build();
```

Comment:
Could you verify how many blobs are usually stored per store? we could use that to inform on this constant

Reply:
Based on the metrics I checked for our jobs (some large jobs, didn't have a way to check avg), it's around 100 blobs per minute for all the stores for a job. So it should be able to hold about 10 mins worth of blob ids without any dedup.